java/client/src/main/java/glide/api/commands/BaseCommands.java

```diff
@@ -1,31 +1,20 @@
 package glide.api.commands;
 
-import glide.ffi.resolvers.RedisValueResolver;
-import glide.managers.BaseCommandResponseResolver;
 import java.util.concurrent.CompletableFuture;
-import response.ResponseOuterClass.Response;
 
-/** Base Commands interface to handle generic command and transaction requests. */
-public interface BaseCommands {
+/**
+ * Base Commands interface to handle generic command and transaction requests.
+ *
+ * @param <T> The data return type.
+ */
+public interface BaseCommands<T> {
```

Comment:
BaseCommands don't have return types.  The RedisValue return type would only be value for cluster mode, standalone mode should have a type t defined. 
We should show how this works for both cluster mode and standalone mode.  Maybe also prototype how this would work with a string handler, which returns a RedisValue<String>...?